pkg/apicast/apicast_option_provider.go

```diff
@@ -228,6 +239,28 @@ func (a *APIcastOptionsProvider) additionalAnnotations() map[string]string {
 		annotations[GatewayConfigurationSecretResverAnnotation] = a.APIcastOptions.GatewayConfigurationSecret.ResourceVersion
 	}
 
+	if a.APIcastOptions.HTTPSCertificateSecret != nil {
+		annotations[HttpsCertSecretResverAnnotation] = a.APIcastOptions.HTTPSCertificateSecret.ResourceVersion
+	}
+
+	if a.APIcastOptions.TracingConfig.Enabled && a.APIcastOptions.TracingConfig.Secret != nil {
+		annotations[OpenTracingSecretResverAnnotation] = a.APIcastOptions.TracingConfig.Secret.ResourceVersion
+	}
+
+	for idx := range a.APIcastOptions.CustomEnvironments {
+		// Secrets must exist
+		// Annotation key includes the name of the secret
+		annotationKey := fmt.Sprintf("%s%s", CustomEnvSecretResverAnnotationPrefix, a.APIcastOptions.CustomEnvironments[idx])
+		annotations[annotationKey] = a.APIcastOptions.CustomEnvironments[idx].ResourceVersion
+	}
+
+	for idx := range a.APIcastOptions.CustomPolicies {
+		// Secrets must exist
+		// Annotation key includes the name of the secret
+		annotationKey := fmt.Sprintf("%s%s", CustomPoliciesSecretResverAnnotationPrefix, a.APIcastOptions.CustomPolicies[idx].Secret.Name)
+		annotations[annotationKey] = a.APIcastOptions.CustomPolicies[idx].Secret.ResourceVersion
+	}
```

Comment:
because the structure is `CustomPolicies                      []CustomPolicy`. It would work if it was `CustomPolicies                      []*CustomPolicy`

Reply:
well, it would work either way, because it is only reading. With your proposal, you are adding an extra structure copy. I am not copying the structure and accessing it through `idx` index.